iXBRLViewerPlugin/constants.py

```diff
@@ -23,13 +23,98 @@
 DEFAULT_VIEWER_PATH = os.path.join(os.path.dirname(__file__), "viewer", "dist", DEFAULT_JS_FILENAME)
 
 FEATURE_CONFIGS = [
+    FeatureConfig(
+        key='home_link_label',
+        label='Home Link (Label)',
+        description='Customizes the label of the "Home" link enabled by "home_link_url".',
+        cliAction='store',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None,
+    ),
+    FeatureConfig(
+        key='home_link_url',
+        label='Home Link (URL)',
+        description='Adds a "Home" link that directs to the specified URL. Label can be customized with "home_link_label".',
+        cliAction='store',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None,
+    ),
+    FeatureConfig(
+        key='support_link',
+        label='Support Link',
+        description='Adds a "Support" link that directs to the specified URL.',
+        cliAction='store',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None,
+    ),
+    FeatureConfig(
+        key='survey_link',
+        label='Survey Link',
+        description='Adds a "Survey" link that directs to the specified URL.',
+        cliAction='store',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None,
+    ),
     FeatureConfig(
         key='review',
         label='Review Mode',
         description='Enables highlighting of untagged numbers and dates.',
+        cliAction='store_true',
+        cliDefault=False,
         guiVisible=True,
         guiDefault=False
+    ),
+    FeatureConfig(
+        key='mandatory_facts',
+        label='Mandatory Facts',
+        description='Enables the display of mandatory facts for the selected taxonomy.',
+        cliAction='store',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None
+    ),
+    FeatureConfig(
+        key='search_on_startup',
+        label='Search on startup',
+        description='Show the search pane by default on startup',
+        cliAction='store_true',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None
+    ),
+    FeatureConfig(
+        key='highlight_facts_on_startup',
+        label='Highlight facts on startup',
+        description='Default "Highlight all facts" to on',
+        cliAction='store_true',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None
     )
 ]
-
 GUI_FEATURE_CONFIGS = [c for c in FEATURE_CONFIGS if c.guiVisible]
+
+MANDATORY_FACTS = {
```

Comment:
It seems surprising to embed this metadata here.

Reply:
Raised https://github.com/Arelle/ixbrl-viewer/issues/767